resources/project.rb

```diff
@@ -123,6 +146,72 @@
     properties["project.#{key}"] = value
   end
 
+  if new_resource.scm_import || new_resource.scm_export
+    directory ::File.join(node['rundeck_server']['datadir'], 'projects', new_resource.name, 'scm') do
+      user     'rundeck'
+      group    'rundeck'
+      mode     '0770'
+    end
+  end
+
+  # configure scm_export
+  if new_resource.scm_export
+    scm_export = {}
+    scm_export['scm.export.type'] = 'git-export'
+    scm_export['scm.export.username'] = 'rundeck'
+    scm_export['scm.export.config.branch'] = 'master'
+    scm_export['scm.export.config.strictHostKeyChecking'] = 'yes'
+    scm_export['scm.export.config.pathTemplate'] = '${job.group}${job.name}-${job.id}.${config.format}'
+    scm_export['scm.export.config.dir'] = ::File.join(node['rundeck_server']['datadir'], 'projects', new_resource.name, 'scm')
+    scm_export['scm.export.config.format'] = 'yaml'
+    scm_export['scm.export.config.importUuidBehavior'] = 'preserve'
+    scm_export['scm.export.config.fetchAutomatically'] = true
+    scm_export['scm.export.enabled'] = true
+    scm_export['scm.export.config.committerEmail'] = '${user.email}'
+    scm_export['scm.export.config.committerName'] = '${user.fullName}'
+    new_resource.scm_export.each do |k, v|
+      scm_export["scm.export.#{k}"] = v
+    end
+
+    template ::File.join(node['rundeck_server']['datadir'], 'projects', new_resource.name, 'etc', 'scm-export.properties') do
+      source   'properties.erb'
+      user     'rundeck'
+      group    'rundeck'
+      mode     '0660'
+      cookbook new_resource.cookbook
+      variables(properties: scm_export)
+    end
+  end
+
+  # configure scm_import
+  if new_resource.scm_import
```

Comment:
This part coyld be factorized with export

Reply:
Can you clarify what do you mean?

Reply:
lines 200-216 are nearly identical to lines 171-187. I suggest we avoid duplicating this code.
For instance:
```
def scm_config(type)
  scm_config = {}
  scm_config["scm.#{type}.username"] = 'rundeck'
  scm_config["scm.#{type}.config.branch"] = 'master'
  ...
  scm_config
end


template ::File.join(node['rundeck_server']['datadir'], 'projects', new_resource.name, 'etc', 'scm-import.properties') do
       source   'properties.erb'
       user     'rundeck'
       group    'rundeck'
       mode     '0660'
       cookbook new_resource.cookbook
       variables(properties: scm_config('import'))
     end

```

Reply:
OK, done